api/src/test/java/io/serverlessworkflow/api/test/ValidationErrorTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2020-Present The Serverless Workflow Specification Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.serverlessworkflow.api.test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import io.serverlessworkflow.api.validation.ValidationError;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import org.junit.jupiter.api.Test;
+
+public class ValidationErrorTest {
+
+  @Test
+  void duplicateTest() {
+
+    Collection<ValidationError> duplicatedErrors = addMessagesToCollection(new ArrayList<>());
+    Collection<ValidationError> errors = addMessagesToCollection(new LinkedHashSet<>());
+
+    assertEquals(duplicatedErrors.size(), 3);
+    assertEquals(errors.size(), 2);
+
+    assertEquals(duplicatedErrors.iterator().next().getMessage(), "This is the first message");
+    assertEquals(errors.iterator().next().getMessage(), "This is the first message");
+  }
+
+  private Collection<ValidationError> addMessagesToCollection(Collection<ValidationError> errors) {
+    ValidationError first = new ValidationError();
```

Comment:
i dont really understand the purpose of this test. in a workflow definition you can have multiple workflow states create same error message. for example you have two states that are missing the name prop or something. 
in this case imho it does not make sense to treat them as same thing...

Reply:
This is a valid use case. But I believe that the validation error will be different between the states, no?

Reply:
i dont think it will be different, for custom validation errors we can generate same message for different states i think

Reply:
i think this test just shows that java Collection class works :) not sure it really shows that during validation our validation CAN produce multiple same validation errors and then show the benefit of the added change

Reply:
Since the list vs set discussion is not really relevant to this PR (I made the change because I believe it was self evidente) I will rollback to List. 
I honestly cannot see the point of adding exactly the same error meessage twice in a collection of validation errors, but I do not think that discussion should defer this PR, which target is to remove unneeded dependencies.